src/transformation/visitors/sourceFile.ts

```diff
@@ -13,7 +13,10 @@ export const transformSourceFileNode: FunctionVisitor<ts.SourceFile> = (node, co
         const [statement] = node.statements;
         if (statement) {
             assert(ts.isExpressionStatement(statement));
-            statements.push(lua.createReturnStatement([context.transformExpression(statement.expression)]));
+            context.pushPrecedingStatements();
+            const expression = context.transformExpression(statement.expression);
+            statements.push(...context.popPrecedingStatements());
+            statements.push(lua.createReturnStatement([expression]));
```

Comment:
Also seems like a  candidate for the push/pop/concat refactor I mentioned earlier